realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SyncedRealmTests.kt

```diff
@@ -804,6 +806,174 @@ class SyncedRealmTests {
+        destination.delete()
+        assertTrue(copy.renameTo(destination))
+
+        Realm.getInstance(config2).use { realm: Realm ->
+            // Check that the copy contains data
+            assertEquals(1, realm.where<SyncColor>().count())
+            realm.executeTransaction {
+                realm.createObject(SyncColor::class.java, ObjectId())
+            }
+            // Check that we can synchronize with the server
+            realm.syncSession.downloadAllServerChanges()
+            realm.syncSession.uploadAllLocalChanges()
+        }
+    }
+
+    @Test
+    fun writeCopyEncryptedTo() {
+        val key = TestHelper.getRandomKey()
+
+        val user = createNewUser()
+        val partitionValue = UUID.randomUUID().toString()
+        val config: SyncConfiguration = createCustomConfig(user, partitionValue)
+        val copy = File(Realm.getApplicationContext()!!.filesDir,"copy.realm")
+        Realm.getInstance(config).use { realm: Realm ->
+            realm.executeTransaction {
+                it.insert(SyncColor())
+            }
+
+            // Create copy of file
+            if (copy.exists()) {
+                copy.delete()
+            }
+            realm.syncSession.uploadAllLocalChanges()
+            realm.syncSession.downloadAllServerChanges() // Work around for https://github.com/realm/realm-core/issues/4865
+            realm.writeEncryptedCopyTo(copy, key)
+        }
+
+        // Manually move the file into the place of another user.
+        // and verify that the user can sync with the Realm.
+        // We cannot use the "assetFile" API that users do, as terminating
+        // Sync renders the file invalid, so we cannot package a test file
+        // for later use.
+        val user2 = createNewUser()
+        val config2: SyncConfiguration = SyncConfiguration.Builder(user2, partitionValue)
+                .schema(SyncColor::class.java)
+                .encryptionKey(key)
+                .build()
+        val destination = File(config2.path)
+        destination.delete()
+        assertTrue(copy.renameTo(destination))
+
+        Realm.getInstance(config2).use { realm: Realm ->
+            // Check that the copy contains data
+            assertEquals(1, realm.where<SyncColor>().count())
+            realm.executeTransaction {
+                realm.createObject(SyncColor::class.java, ObjectId())
+            }
+
+            // Check that we can synchronize with the server
+            realm.syncSession.downloadAllServerChanges()
+            realm.syncSession.uploadAllLocalChanges()
+        }
+    }
+
+    @Test
+    fun assetFile() {
+        val user = createNewUser()
+        val config = SyncConfiguration.Builder(user, "e873fb25-11ef-498f-9782-3c8e1cd2a12c")
+            .assetFile("synced_realm_e873fb25-11ef-498f-9782-3c8e1cd2a12c_no_client_id.realm")
+            .schema(SyncColor::class.java)
+            .build()
+
+        Realm.getInstance(config).use { realm: Realm ->
```

Comment:
I think this also needs to verify you can call `SyncSession.uploadAllLocalChanges()` ... at least that was one of the bugs in Core, that you could open the file locally, but you could not sync with it.

Reply:
I removed `uploadAllLocalChanges()` because I though it was not supported to run it at that point, but it's a deeper issue around how we test `assetFile()`.

The sync Realm `writeCopy` file is no longer bound to the user, but it is still to the server instance: the asset file we provide only works on the instance it was created and will not work on other instances, thats why `uploadAllLocalChanges` fails.

We might need to mock the assets to be able to pass a freshly generated one into the config.